common/mission_planner_rviz_plugin/src/route_selector_panel.cpp

```diff
@@ -0,0 +1,136 @@
+    default:                       return "-----";
+  }
+  // clang-format on
+}
+
+RouteSelectorPanel::RouteSelectorPanel(QWidget * parent) : rviz_common::Panel(parent)
+{
+  main_state_ = new QLabel("unknown");
+  main_clear_ = new QPushButton("clear");
+  mrm_state_ = new QLabel("unknown");
+  mrm_clear_ = new QPushButton("clear");
+  planner_state_ = new QLabel("unknown");
+
+  connect(main_clear_, &QPushButton::clicked, this, &RouteSelectorPanel::onMainClear);
+  connect(mrm_clear_, &QPushButton::clicked, this, &RouteSelectorPanel::onMrmClear);
+
+  const auto layout = new QGridLayout();
+  setLayout(layout);
+  layout->addWidget(new QLabel("main"), 0, 0);
+  layout->addWidget(main_state_, 0, 1);
+  layout->addWidget(main_clear_, 0, 2);
+  layout->addWidget(new QLabel("mrm"), 1, 0);
+  layout->addWidget(mrm_state_, 1, 1);
+  layout->addWidget(mrm_clear_, 1, 2);
+  layout->addWidget(new QLabel("planner"), 2, 0);
+  layout->addWidget(planner_state_, 2, 1);
+}
+
+void RouteSelectorPanel::onInitialize()
+{
+  auto lock = getDisplayContext()->getRosNodeAbstraction().lock();
+  auto node = lock->get_raw_node();
+
+  const auto durable_qos = rclcpp::QoS(1).transient_local();
+
+  sub_main_goal_ = node->create_subscription<PoseStamped>(
+    "/rviz/route_selector/main/goal", rclcpp::QoS(1),
+    std::bind(&RouteSelectorPanel::onMainGoal, this, std::placeholders::_1));
+  sub_mrm_goal_ = node->create_subscription<PoseStamped>(
+    "/rviz/route_selector/mrm/goal", rclcpp::QoS(1),
+    std::bind(&RouteSelectorPanel::onMrmGoal, this, std::placeholders::_1));
+  sub_main_state_ = node->create_subscription<RouteState>(
+    "/planning/mission_planning/route_selector/main/state", durable_qos,
+    std::bind(&RouteSelectorPanel::onMainState, this, std::placeholders::_1));
+  sub_mrm_state_ = node->create_subscription<RouteState>(
+    "/planning/mission_planning/route_selector/mrm/state", durable_qos,
+    std::bind(&RouteSelectorPanel::onMrmState, this, std::placeholders::_1));
+  sub_planner_state_ = node->create_subscription<RouteState>(
+    "/planning/mission_planning/state", durable_qos,
+    std::bind(&RouteSelectorPanel::onPlannerState, this, std::placeholders::_1));
+
+  cli_main_clear_ =
+    node->create_client<ClearRoute>("/planning/mission_planning/route_selector/main/clear_route");
+  cli_mrm_clear_ =
+    node->create_client<ClearRoute>("/planning/mission_planning/route_selector/mrm/clear_route");
+  cli_main_set_waypoint_ = node->create_client<SetWaypointRoute>(
+    "/planning/mission_planning/route_selector/main/set_waypoint_route");
+  cli_mrm_set_waypoint_ = node->create_client<SetWaypointRoute>(
+    "/planning/mission_planning/route_selector/mrm/set_waypoint_route");
+}
+
+void RouteSelectorPanel::onMainGoal(const PoseStamped::ConstSharedPtr msg)
+{
+  const auto req = std::make_shared<SetWaypointRoute::Request>();
+  req->header = msg->header;
+  req->goal_pose = msg->pose;
+  req->allow_modification = true;
```

Comment:
it is better to be able to set true/false.
but it seems enough for current project usecase

Reply:
This is not supported as it is for confirming switching with MRM. For goal modification, the `tier4_adapi_rviz_plugin::RoutePanel` in the is available.